importer.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-exec/tfexec"
+)
+
+type Importer struct {
+	e *tfexec.Terraform
+	c *tfe.Client
+}
+
+func NewImporter(tf *tfexec.Terraform, token string, host string) (*Importer, error) {
+	tfc, err := tfe.NewClient(&tfe.Config{
+		Token:   token,
+		Address: fmt.Sprintf("https://%s", host),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &Importer{
+		tf,
+		tfc,
+	}, nil
+}
+
+func (tf *Importer) ImportWorkspace(ctx context.Context, name string, organization string, opts ...tfexec.ImportOption) error {
+	address := fmt.Sprintf("tfe_workspace.workspace[\"%s\"]", name)
+
+	state, err := tf.e.Show(context.Background())
+	if err != nil {
+		log.Fatalf("Failed to read state: %s", err)
+	}
+
+	found := false
+	if state.Values == nil {
+		fmt.Println("No state found, importing")
```

Comment:
Careful mixing around UI here, it'll make things harder to test eventually. Consider splitting out methods, e.g. `ShouldImport(...) (bool, error)` that are UI-less so the main func or other high level funcs can call them and then interleave UI output.